### ISSUE TITLE
Rule Change: (`no-console`) warn about `import { log } from "console"`

### ISSUE THREAD

**Issue by Zwyx** [eslint/eslint#17470]
### What rule do you want to change?

no-console

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
import { log } from "console";
```

### What does the rule currently do for this code?

Nothing.

### What will the rule do after it's changed?

Warn in the same way it does when writing `console.log`.
It could either warn on the line of the `import`, or on line where log is used (`log(...)`).

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Thank you for making this excellent tool!

**Comment by Rec0iL99** [comment]
Hi @Zwyx, thanks for the issue. I couldn't find any previous discussion around this for the `no-console` rule, so I'm going to push this forward for the team to decide.

**Comment by fasttime** [comment]
`"console"` as a module specifier is peculiar to Node.js: it is not recognized by browsers. Core rules that are specific to Node.js are generally deprecated since ESLint 7 in favor of custom rules like those in [`eslint-plugin-n`](https://www.npmjs.com/package/eslint-plugin-n), so I'm not sure if we want to update the `no-console` rule to account for this Node.js specific case.

If we decide to update the rule, I think a problem should be reported only on the line with the import for consistency with the current implementation, considering for example:

```js
/* eslint no-console: warn */

const log = console.log; // warning
log(1);                  // no warning
log("foo");              // no warning
```

[**DEMO**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWNvbnNvbGU6IHdhcm4gKi9cblxuY29uc3QgbG9nID0gY29uc29sZS5sb2c7IC8vIHdhcm5pbmdcbmxvZygxKTsgICAgICAgICAgICAgICAgICAvLyBubyB3YXJuaW5nXG5sb2coXCJmb29cIik7ICAgICAgICAgICAgICAvLyBubyB3YXJuaW5nIiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==)

Also, I don't think that the default behavior of the rule should be changed. If this suggestion is implemented, it would need a new option.

**Comment by Zwyx** [comment]
Thank you guys for replying. I understand your view. I'll describe my use case, maybe it will help you take the decision:

I created a function called `log`, which filters what it receives depending of the selected log level.

This function is in a file called `utils.ts`.

I once made the mistake of importing `log` from `console` instead of from `./utils.ts`, and I was surprised that ESLint didn't pick it up.

Now, I realised that using `log` leads to confusion, so I renamed this function to `lg`. So my problem is solved.

**Comment by fasttime** [comment]
Thanks for adding context @Zwyx.

> I once made the mistake of importing `log` from `console` instead of from `./utils.ts`, and I was surprised that ESLint didn't pick it up.

Maybe you could use the [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports) rule to catch this error ([**DEMO**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtaW1wb3J0czogW2Vycm9yLCB7XG4gICAgbmFtZTogY29uc29sZSxcbiAgICBtZXNzYWdlOiBQbGVhc2UgdXNlICd1dGlscy50cycgaW5zdGVhZC5cbn1dICovXG5cbmltcG9ydCB7IGxvZyB9IGZyb20gXCJjb25zb2xlXCI7Iiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==)).

**Comment by Zwyx** [comment]
Ah nice! And thanks for the demo. Great idea indeed :slightly_smiling_face:

**Comment by mdjermanovic** [comment]
Closing as this is covered by the `no-restricted-imports` rule, and I agree with @fasttime that the `no-console` rule, which warns about using global variable `console`, shouldn't be updated to account for environment-specific ways to access console.
